src/Public/Remove-CCMStaleDeployment.ps1

```diff
@@ -29,7 +29,11 @@ function Remove-CCMStaleDeployment {
         #>
         $badStates = @(0, 2, 8)
         if ($PSCmdlet.ShouldProcess("$Deployment", "DELETE")) {
-            Get-CCMDeployment -All | Where-Object { $_.CreationDate -ge (Get-Date).AddDays(-$Age) -and $null -eq $_.StartDateTimeUtc -and $_.Result -in $badStates } | Remove-CCMDeployment
+            Get-CCMDeployment | Where-Object {
+                $_.CreationDate -ge (Get-Date).AddDays(-$Age) -and
+                $null -eq $_.StartDateTimeUtc -and
+                $_.Result -in $badStates
+            } | Remove-CCMDeployment

```

Comment:
I think this can be (slightly) simplified and maybe we want to reorder these a _little_ to avoid unnecessary calculations where we can early-discard based on some of these criteria. Also... `-gt` a date in the past means we're only selecting things _younger_ than the desired $Age. That doesn't feel right to me?

```suggestion
            $staleDate = (Get-Date).AddDays(-$Age)
            Get-CCMDeployment | Where-Object {
                -not $_.StartDateTimeUtc -and
                $_.Result -in $badStates -and
                $_.CreationDate -le $staleDate
            } | Remove-CCMDeployment

```

Also... are we sure we want to only prune deployments that have never started?

Reply:
I am very happy to change this logic - the commit I made was a minimal "the command is literally broken right now, and I want tests" - I have a whole branch on a refactor with some improvements, but that wasn't a part of _this_ PR.